pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -139,7 +139,8 @@ def restore(self, checkpoint_path: Optional[_PATH] = None) -> None:
         self.restore_model()
 
         # restore callback states
-        self.restore_callbacks()
+        if self.trainer.state.fn != TrainerFn.TUNING:
```

Comment:
I'm confused actually. The other place `restore_callabacks` is called uses `fit`: https://github.com/PyTorchLightning/pytorch-lightning/blob/3f78c4ca7a89fed8f79d9015e9510f0a0dcc46b5/pytorch_lightning/trainer/trainer.py#L1101

But why would callbacks not be restored for all trainer functions?

Reply:
because we clear out all the callbacks while tuning and use checkpoint connector to restore states of everything (including the callbacks). The saved checkpoint is restored before reloading the params.

https://github.com/PyTorchLightning/pytorch-lightning/blob/3f78c4ca7a89fed8f79d9015e9510f0a0dcc46b5/pytorch_lightning/tuner/lr_finder.py#L207-L237

An alternate solution can be to restore the params and load the checkpoint.